src/main/java/redis/clients/jedis/ShardedJedisPool.java

```diff
@@ -77,32 +77,27 @@ public ShardedJedisFactory(List<JedisShardInfo> shards, Hashing algo, Pattern ke
     public void destroyObject(PooledObject<ShardedJedis> pooledShardedJedis) throws Exception {
       final ShardedJedis shardedJedis = pooledShardedJedis.getObject();
       for (Jedis jedis : shardedJedis.getAllShards()) {
-        try {
-          try {
-            jedis.quit();
-          } catch (Exception e) {
-
-          }
-          jedis.disconnect();
-        } catch (Exception e) {
-
-        }
+        Jedis.destroy(jedis);
       }
     }
 
     @Override
     public boolean validateObject(PooledObject<ShardedJedis> pooledShardedJedis) {
-      try {
-        ShardedJedis jedis = pooledShardedJedis.getObject();
-        for (Jedis shard : jedis.getAllShards()) {
-          if (!shard.ping().equals("PONG")) {
-            return false;
-          }
+      ShardedJedis jedis = pooledShardedJedis.getObject();
+      for (JedisShardInfo shardInfo : jedis.getDistinctShardInfo()) {
+        Jedis shard = jedis.getShard(shardInfo);
+        boolean valid = true;
+        try {
+          valid = shard.ping().equals("PONG");
+        } catch(Exception ex) {
+          valid = false;
+        }
+        if (!valid) {
+          jedis.replaceShard(shardInfo);
```

Comment:
Newly created Jedis instance may fail to ping, or it even throws any Exception while creating Jedis which `validateObject()` doesn't want to.
I think we should take care of that (drop the whole instance and return false), since `validateObject()` should ensure the instance is good.